directory/onion_directory.py

```diff
@@ -5,11 +5,10 @@
 import threading            # for one thread per TCP connection
 
 
-
 #   TODO:
 #    - incorporate encryption key mgmt
-
-
+#    - directory ping of nodes causes routers to fail, must start directory
+#      before router
```

Comment:
this wasn't the case for me

Reply:
it is, when the error occurred I rolled back to your latest change on `onion_router.py` 645c96cca878fa63bf5a07826560de6446f32856

It crashes there too, I assumed that's why you opened issue #19

Reply:
embarassing.  We'll have to handle that. nice find.